pkg/resources/deviceSelectors.go

```diff
@@ -76,8 +78,38 @@ type pfNameSelector struct {
 func (s *pfNameSelector) Filter(inDevices []types.PciNetDevice) []types.PciNetDevice {
 	filteredList := make([]types.PciNetDevice, 0)
 	for _, dev := range inDevices {
-		if contains(s.pfNames, dev.GetPFName()) {
-			filteredList = append(filteredList, dev)
+		selector := getItem(s.pfNames, dev.GetPFName())
+		if selector != "" {
+			if strings.Contains(selector, "#"){
+				// Selector does contain VF index in next format:
+				// <PFName>#<VFIndexStart>-<VFIndexEnd>
+				// In this case both <VFIndexStart> and <VFIndexEnd>
+				// are included in range, for example: "netpf0#3-5"
+				// The VFs 3,4 and 5 of teh PF 'netpf0' will be included
+				// in selector pool
+				fields := strings.Split(selector,"#")
+				if len(fields) != 2 {
+					return filteredList
```

Comment:
what about `continue` instead of `return`?
`continue` allows the rest of `inDevices` be evaluated even there is a format error in one of `pfNames` selector list.

Reply:
It was done intentionally.
Lets say that you have 10 PF names in selector and only in 7-th you have typo, than
it is more easily to detect the error if you see the result is cut at 7-th PF name, then
look for absence of 7-th PF name parsing.
Anyway, I fix thus issue by continuing and printing error message.